bench/Bench/Utils.hs

```diff
@@ -0,0 +1,176 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE InstanceSigs               #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TupleSections              #-}
+
+module Bench.Utils (
+    -- * Environment types
+    BenchEnv (..)
+    -- * Initialisation and cleanup for environments
+  , initBenchEnv
+  , cleanupBenchEnv
+    -- * Defaults for LMDB Limits
+  , simpleLMDBLimits
+    -- * Defaults for database population
+  , noPopulateDb
+  , populateDbUnique
+    -- * Default types for keys and values
+  , Key (..)
+  , Value (..)
+  , keySize
+  , key0
+  , pKey
+  , pValue
+  , valueSize
+  ) where
+
+import           Codec.Serialise
+import           Control.DeepSeq
+import           Control.Monad
+import           Control.Monad.IO.Class
+import qualified Data.ByteString.Short         as B (ShortByteString, pack)
+import           Data.Proxy
+import           GHC.Generics                  (Generic)
+import           System.Directory
+import           System.IO.Temp
+
+import           Test.Tasty.QuickCheck
+
+import           Database.LMDB.Simple
+import qualified Database.LMDB.Simple.Internal as Internal (putNoOverwrite)
+
+{-------------------------------------------------------------------------------
+  Environment types
+-------------------------------------------------------------------------------}
+
+data BenchEnv k v = BenchEnv {
+    dbFilePath :: FilePath
+  , dbEnv      :: Environment ReadWrite
+  , db         :: Database k v
+  }
+
+-- TODO: is this instance sufficient? I think so, since a @'BenchEnv'@ mainly
```

Comment:
I know the answer: they are mostly pointers. I've updated the comment to suggest improvements:
```haskell
-- TODO: The @'dbEnv'@ and @'db'@ fields of @'BenchEnv'@ consist mainly of
-- pointers, so we can omit them here because they will be likely not to
-- contribute much to evaluation time. If we want to evaluate these fields
-- instead, we should change the lower-level Haskell bindings to expose more
-- constructors.
instance NFData (BenchEnv k v) where
```